src/MapBuilding.js

```diff
@@ -1,33 +1,97 @@
+  const { buildingGeoJSON, waterLabelColor, selectedAddress, surroundingBuildings, labelColors } = props;
   const polygonBounds = getPolygonBounds(buildingGeoJSON);
   const polygonCenter = getPolygonCenter(polygonBounds);
-  // zoomlevel 19 is the most zoomed in where you still get a basemap
+  // Zoomlevel 19 is the most zoomed in where you still get a basemap
   const zoomLevel = 19;
   let position = [polygonCenter.yMean, polygonCenter.xMean];
+  // Returning the getBounds of a react-leaflet map is quite complicated,
+  // so I went for an easier solution: substracting and adding from the
+  // x and y of the center of the map to get the bounding box for
+  // the api call for the surrounding buildings.
+  const bbox = createBbox(polygonCenter.xMean, 0.001, polygonCenter.yMean, 0.00075);
   let latlngs = [];
+  const [buildingCoordsAndLabels, setBuildingCoordsAndLabels] = useState([]);
 
-  if (buildingGeoJSON) {
-    for (var i = 0; i < buildingGeoJSON.coordinates[0].length; i++) {
-      // GeoJSON has longitude first
-      let lng = buildingGeoJSON.coordinates[0][i][0];
-      let lat = buildingGeoJSON.coordinates[0][i][1];
-      // Polygon of react-leaflet has latitude first
-      latlngs.push([lat, lng]);
-    }
+  for (var i = 0; i < buildingGeoJSON.coordinates[0].length; i++) {
+    // GeoJSON has longitude first
+    let lng = buildingGeoJSON.coordinates[0][i][0];
+    let lat = buildingGeoJSON.coordinates[0][i][1];
+    // Polygon of react-leaflet has latitude first
+    latlngs.push([lat, lng]);
   }
 
+  useEffect(() => {
+    fetch(
+      `/api/v2/buildings/?in_bbox=${bbox}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    .then(function(response) {
+      if (response.status !== 200) {
+        throw new Error("building get status not 200: " + response.status);
+      }
+      return response.json();
+    })
+    .then(function(parsedJSON) {
+      // A maximum of a 100 results will be given back by the api.
+      let buildingCoordsAndLabels = parsedJSON.results.map((building) => {
+        let invertedCoords = building.geometry.coordinates[0].map((coord) => {
+          // GeoJSON has longitude first
+          let lng = coord[0];
+          let lat = coord[1];
+          // Polygon of react-leaflet has latitude first
+          return ([lat, lng]);
+        });
+        // If the invertedCoords are the same as latlngs (of the selected
+        // building), make the selected building of the surrounding buildings
+        // transparent. To prevent from the selected building being drawn
+        // twice.
+        if (arraysOfCoordsEqual(latlngs, invertedCoords)){
+          // Make the selected building transparent in the surrounding buildings
+          return ({ coords: invertedCoords, color: "rgba(0, 0, 0, 0" });
+        } else {
+          let waterlabel = "G";  // Default and worst label
```

Comment:
This should really be defined somewhere, not just be a magic constant in the code.